src/main/java/com/squareup/javapoet/TypeVariableName.java

```diff
@@ -57,6 +57,18 @@ private TypeVariableName(String name, List<TypeName> bounds, List<AnnotationSpec
     return new TypeVariableName(name, bounds);
   }
 
+  public TypeVariableName withBounds(Type... bounds) {
+    return get(name, bounds);
```

Comment:
Aren't bounds supposed to have a class first (if any) followed by interfaces?
https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.4